internal/js/compiler/compiler.go

```diff
@@ -115,22 +115,28 @@ func (ps *parsingState) parseImpl(src, filename string, commonJSWrap bool) (*ast
 		return prg, code, nil
 	}
 
-	if strings.HasSuffix(filename, ".ts") {
+	isTsExtensionFile := strings.HasSuffix(filename, ".ts")
+	isStdin := filename == "file:///-"
+	if !isTsExtensionFile && !isStdin {
+		return nil, "", err
+	}
+
+	if isTsExtensionFile {
```

Comment:
I am not certain we should report the stdin parsing as well ... if we are going to do it. 

I guess we can only report it past the `StripTypes` if the worry is that this might not work and we will report stuff more?